### PR TITLE
map column

### PR DESCRIPTION
**Pull request by fabrizio-m**
makes generic a few trivial implementations and adds the method `map_column` to all expressions.

### CODE REVIEW

**Comment by fabrizio-m** [comment]
!ci-build-me

File: kimchi/src/circuits/expr.rs [hunk 1]
```diff
@@ -622,8 +622,45 @@ pub enum Expr<C, Col: ColTrait = Column> {
...
 }
 
-impl<C: Zero + One + Neg<Output = C> + PartialEq + Clone> Expr<C> {
-    fn apply_feature_flags_inner(&self, features: &FeatureFlags) -> (Expr<C>, bool) {
+impl<C, Col> Expr<C, Col>
+where
+    C: Zero + One + Neg<Output = C> + PartialEq + Clone,
+    Col: ColTrait,
+{
+    ///maps expressiosn over some columns to expressions over other columns
```
Comment (querolita) [inline kimchi/src/circuits/expr.rs:630]: nit expressions

**Review by querolita — APPROVED** [review]

File: kimchi/src/circuits/expr.rs [hunk 2]
```diff
@@ -622,8 +622,45 @@ pub enum Expr<C, Col: ColTrait = Column> {
...
+                Expr::Cell(Variable { col: map(col), row })
+            }
+            Expr::Double(exp) => Expr::Double(Box::new((*exp).map_column(map))),
+            Expr::Square(exp) => Expr::Square(Box::new((*exp).map_column(map))),
+            Expr::BinOp(op, exp1, exp2) => {
+                let exp1 = exp1.map_column(map);
+                let exp2 = exp2.map_column(map);
+                Expr::BinOp(op, Box::new(exp1), Box::new(exp2))
+            }
+            Expr::VanishesOnLast4Rows => Expr::VanishesOnLast4Rows,
```
Comment (querolita) [inline kimchi/src/circuits/expr.rs:649]: Again, here this expression no longer exists in develop. Please get the most updated develop (or wait until we know what to do regarding `develop` vs `master`)

**Review by querolita — CHANGES_REQUESTED** [review]

**Comment by querolita** [comment]
Just a reminder -> I believe we are merging against master instead of develop

**Comment by volhovm** [comment]
Closing, as per @fabrizio-m most of what is useful was already integrated into folding.

**Pull request closed without merging**